website/scripts/index_search_content.js

```diff
@@ -1,3 +1,81 @@
-const { indexDocsContent } = require('@hashicorp/react-search/tools')
+require('dotenv').config()
+const fs = require('fs')
+const path = require('path')
+const {
+  indexContent,
+  getDocsSearchObject,
+} = require('@hashicorp/react-search/tools')
+const resolveNavData = require('../components/remote-plugin-docs/utils/resolve-nav-data')
+const fetchGithubFile = require('../components/remote-plugin-docs/utils/fetch-github-file')
 
-indexDocsContent()
+// Run indexing
+indexContent({ getSearchObjects })
+
+async function getSearchObjects() {
```

Comment:
I've used `indexContent` here per Will's suggestion, and it feels a _lot_ cleaner than the previous `indexDocsContent` approach 💖 

The idea here is to index all content referenced in `nav-data.json` files. This includes resolved remote plugin files. Hopefully this feels like it gives us a mix of the benefits - we only index pages that will actually be rendered (rather than indexing all content files as was done previously), but the indexing process is still separate from the build process.